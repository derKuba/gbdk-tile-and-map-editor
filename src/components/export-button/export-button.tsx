import { Component, Host, h, Prop } from '@stencil/core';

import Tunnel from '../../provider/StateProvider';

@Component({
    tag: 'mvjp-export-button',
    shadow: true,
    styleUrl: 'export-button.css',
})
export class ExportButton {
    /**
     *  display modal toggle event
     */
    @Prop() setIsModalOpen: Function;

    private openMenu = () => {
        this.setIsModalOpen(true);
    };

    render() {
        return (
            <Host>
                <button class="export-button" onClick={this.openMenu}>
                    Export
                </button>
            </Host>
        );
    }
}

Tunnel.injectProps(ExportButton, ['setIsModalOpen']); // eslint-disable-line @stencil/ban-side-effects
