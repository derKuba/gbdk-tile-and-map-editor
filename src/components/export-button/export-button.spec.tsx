import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { ExportButton } from './export-button';

describe('mvjp-export-button', () => {
    it('builds', () => {
        expect(new ExportButton()).toBeTruthy();
    });
    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [ExportButton],
            template: () => {
                return <mvjp-export-button> </mvjp-export-button>;
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('should test clickhandler', async () => {
        const setIsModalOpen = jest.fn();

        const page = await newSpecPage({
            components: [ExportButton],
            template: () => {
                return <mvjp-export-button setIsModalOpen={setIsModalOpen}> </mvjp-export-button>;
            },
        });

        page.root.shadowRoot.querySelector('button').click();
        await page.waitForChanges();
        expect(setIsModalOpen).toBeCalled();
    });
});
