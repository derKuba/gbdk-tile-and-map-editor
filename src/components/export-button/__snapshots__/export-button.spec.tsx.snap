// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-export-button matches snapshot 1`] = `
<mvjp-export-button>
  <mock:shadow-root>
    <button class="export-button">
      Export
    </button>
  </mock:shadow-root>
</mvjp-export-button>
`;
